tile/targets/cpu/compiler.cc

```diff
@@ -1846,16 +1846,47 @@ void Compiler::OutputBool(llvm::Value* ret, const stripe::Intrinsic& intrinsic)
   ret->setName(intrinsic.outputs[0]);
 }
 
-void Compiler::CallIntrinsicFunc(const stripe::Intrinsic& stmt, const char* name_f32, const char* name_f64) {
-  assert(1 == stmt.inputs.size());
-  Scalar op = Cast(scalars_[stmt.inputs[0]], stmt.type);
-  std::vector<llvm::Value*> argvals{op.value};
+void Compiler::CallIntrinsicFunc(const stripe::Intrinsic& stmt, const char* name_f32, const char* name_f64,
+                                 const size_t numParamsIn) {
+  size_t numParams = stmt.inputs.size();
+  switch (numParamsIn) {
+    case 1:
+      if (numParams != 1) {
+        throw std::runtime_error("CallIntrinsicFunction expects 1 parameter");
+      }
+      break;
+    case 2:
+      if (numParams != 2) {
+        throw std::runtime_error("CallIntrinsicFunction expects 2 parameters");
+      }
+      break;
+    default:
+      throw std::runtime_error("CallIntrinsicFunction expects 1 or 2 parameters");
+  }
+
+  Scalar op1 = Cast(scalars_[stmt.inputs[0]], stmt.type);
+  Scalar op2;
```

Comment:
It makes me nervous that there's an uninitialized var, seems like we could generically just cast all the params? like:

```
std::vector<llvm::Value*> argvals;
for (auto operand : stmt.inputs) {
  argvals.emplace_back(Cast(scalars_[operand], stmt.type);
}
```

Reply:
The types will work nicely in this case (always the same) - that is the nature of pow(). But there might be cases where special attention is needed for the types. I'd rather force a manual examination instead of automatic buggeneration, where figuring out what happens would be a mistery.